tools/src/tester/ChapterReader.ts

```diff
@@ -54,18 +54,18 @@ export default class ChapterReader {
       if (payload !== undefined) response.payload = payload
       this.logger.info(`<= ${r.status} (${r.headers['content-type']}) | ${to_json(response.payload)}`)
     }).catch(e => {
-      if (e.response == null) {
+      if (e.response === undefined) {
```

Comment:
Just curious. Maybe I've gotten it wrong.  Isn't `== null` the same as `=== undefined || ===null` and `==null` is still valid in this case?

Reply:
I am just too scared of `== null`, it can only be undefined here AFAIK.

Reply:
I think `== null` is favorable in this case, and most situations. Consider these 2 scenarios:
```ts
const x = {}
x.foo = null
console.log(x.foo) // null
console.log(x.foo == null) // true
console.log(x.foo === undefined) // false
```
```ts
const x= {}
consolee.log(x.foo) // undefined
console.log(x.foo == null) // true
console.log(x.foo === undefined) // true
```

As you can see, if `x.foo` is explicitly set to `null`, then `=== undefined` will fail, while `== null` accounts for both whether `x.foo` is unset or explicitly set to `null`